var/spack/repos/builtin/packages/py-pyside2/package.py

```diff
@@ -50,13 +63,20 @@ class PyPyside2(PythonPackage):
         description="Enables the generation of html and man page documentation",
     )
 
-    depends_on("python@2.7.0:2.7,3.5.0:3.5,3.6.1:", type=("build", "run"))
-    depends_on("python@2.7.0:2.7,3.5.0:3.5,3.6.1:3.8", when="@:5.14", type=("build", "run"))
+    # see https://wiki.qt.io/Qt_for_Python#Python_compatibility_matrix
+    depends_on("python@2.7.0:2.7,3.5.0:3.5,3.6.1:3.8", when="@:5.15.0", type=("build", "run"))
+    depends_on(
+        "python@2.7.0:2.7,3.5.0:3.5,3.6.1:3.9", when="@5.15.1:5.15.7", type=("build", "run")
+    )
+    depends_on("python@2.7:3.10", when="@5.15.8", type=("build", "run"))
+    depends_on("python@3.5:3.10", when="@5.15.9:5.15.10", type=("build", "run"))
+    depends_on("python@3.6:3.11", when="@5.15.11:5.15.15", type=("build", "run"))
```

Comment:
If you want, we can simplify the Python versioning here, since Spack officially supports installing Python 3.7 or later.

Reply:
Both is fine for me. The current version is just the table from https://wiki.qt.io/Qt_for_Python#Python_compatibility_matrix written out as text.

If you prefer the shorter version (better for the dependency solver?), I can simplify it!

Reply:
Let's simplify this in a following PR.